modules/tinymce/src/plugins/table/main/ts/ui/CellDialog.ts

```diff
@@ -5,90 +5,68 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
-import { HTMLTableCellElement, Node } from '@ephox/dom-globals';
-import { Fun } from '@ephox/katamari';
+import { HTMLTableCellElement } from '@ephox/dom-globals';
+import { Fun, Arr } from '@ephox/katamari';
 import Editor from 'tinymce/core/api/Editor';
-import Tools from 'tinymce/core/api/util/Tools';
 import * as Styles from '../actions/Styles';
 import * as Util from '../alien/Util';
 import { hasAdvancedCellTab } from '../api/Settings';
 import * as CellDialogGeneralTab from './CellDialogGeneralTab';
 import * as Helpers from './Helpers';
 import { DomModifier } from './DomModifier';
 import { Types } from '@ephox/bridge';
+import { Selections } from '../selection/Selections';
+import * as TableSelection from '../selection/TableSelection';
+import * as CellOperations from '../queries/CellOperations';
 
 type CellData = Helpers.CellData;
 
-const updateSimpleProps = (modifiers, data: CellData) => {
-  modifiers.setAttrib('scope', data.scope);
-  modifiers.setAttrib('class', data.class);
-  modifiers.setStyle('width', Util.addSizeSuffix(data.width));
-  modifiers.setStyle('height', Util.addSizeSuffix(data.height));
+const updateSimpleProps = (modifier: DomModifier, data: CellData) => {
+  modifier.setAttrib('scope', data.scope);
+  modifier.setAttrib('class', data.class);
+  modifier.setStyle('width', Util.addSizeSuffix(data.width));
+  modifier.setStyle('height', Util.addSizeSuffix(data.height));
 };
 
-const updateAdvancedProps = (modifiers, data: CellData) => {
-  modifiers.setStyle('background-color', data.backgroundcolor);
-  modifiers.setStyle('border-color', data.bordercolor);
-  modifiers.setStyle('border-style', data.borderstyle);
-  modifiers.setStyle('border-width', Util.addSizeSuffix(data.borderwidth));
+const updateAdvancedProps = (modifier: DomModifier, data: CellData) => {
+  modifier.setStyle('background-color', data.backgroundcolor);
```

Comment:
How come these don't use the new formats? It seems that this really should be using it, otherwise we have branched logic which will likely end up with inconsistencies and maintenance issues in future.

Reply:
If that's the cause, then the rest of the code shouldn't be using the formatter and should be using this logic instead.

Also yeah, last I recall we discussed it, we considered the `IfTruthy` thing a bug for these styles, as it prevents you from removing the styles.

Reply:
I guess we could treat it as tech debt, but I'd be **very** concerned it'd never happen and then we'd end up with this mess forever.

Reply:
I have refactored the code so that the cell style command and cell dialog are now consistent, using the new formats as well as the same cell selection logic. I am still a bit unsure on how we can fix/improve the current DomModifer behaviour so haven't touched that yet.

Reply:
I think I like this approach now 🤔 I'm sure we can optimise the whole applying table styles thing more later (e.g. deal with DomModifier), but this works for me for now.